docs/concepts/layout.md

```diff
@@ -0,0 +1,48 @@
+# Layout
```

Comment:
@ishanarya0 lets also add an architecture doc that explains the overall architecture and how things fit together. A good reference example would be https://github.com/hashicorp/waypoint/blob/main/ARCHITECTURE.md

Reply:
@ishanarya0 Let's also add a guide on how developers can create a new module.